test/unit/product/product-component.js

```diff
@@ -1043,113 +1048,230 @@ describe('Product Component class', () => {
+        describe('when button destination is not modal', () => {
+          beforeEach(() => {
+            product.config.product.buttonDestination = 'cart';
+            Object.defineProperties(product, {
+              buttonActionAvailable: {
+                writable: true,
+              },
+              variantExists: {
+                writable: true,
+              },
+              variantInStock: {
+                writable: true,
+              },
+            });
           });
 
-          describe('if variant is in stock', () => {
-            it('returns true', () => {
-              assert.ok(product.buttonEnabled);
-            });
+          it('returns false if button action is not available', () => {
+            product.buttonActionAvailable = false;
+            product.variantExists = true;
+            product.variantInStock = true;
+            assert.isFalse(product.buttonEnabled);
           });
 
-          describe('if variant is not in stock', () => {
-            it('returns false', () => {
-              product.selectedVariant = {
-                available: false,
-              };
-              assert.notOk(product.buttonEnabled);
-            });
+          it('returns false if variant does not exist', () => {
+            product.buttonActionAvailable = true;
+            product.variantExists = false;
+            product.variantInStock = true;
+            assert.isFalse(product.buttonEnabled);
+          });
+
+          it('returns false if variant is not in stock', () => {
+            product.buttonActionAvailable = true;
+            product.variantExists = true;
+            product.variantInStock = false;
+            assert.isFalse(product.buttonEnabled);
+          });
+
+          it('returns true if button action is available, product exists, and product is in stock', () => {
+            product.buttonActionAvailable = true;
+            product.variantExists = true;
+            product.variantInStock = true;
+            assert.isTrue(product.buttonEnabled);
           });
         });
       });
 
       describe('variantExists', () => {
-        describe('if variant exists for selected options', () => {
-          it('returns true', async () => {
-            await product.init(testProductCopy);
-            product.selectedVariant = {id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8xMjM0NQ=='};
-            assert.isOk(product.variantExists);
-          });
+        beforeEach(async () => {
+          await product.init(testProductCopy);
         });
 
-        describe('if variant does not exist for selected options', () => {
-          it('returns false', async () => {
-            await product.init(testProductCopy);
-            product.selectedVariant = null;
-            assert.isNotOk(product.variantExists);
-          });
+        it('returns true if variant exists in model', async () => {
+          product.selectedVariant = {id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8xMjM0NQ=='};
```

Comment:
Assuming this id string is coming from the fixture, are we able to reference the fixture value instead of this string? That will make it easier to understand where this string is coming from.